lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -231,16 +231,17 @@ export class AuthorizationCodeClient extends BaseClient {
             parameterBuilder.addPrompt(request.prompt);
         }
 
-        if (request.loginHint) {
-            parameterBuilder.addLoginHint(request.loginHint);
-        }
-
         if (request.domainHint) {
             parameterBuilder.addDomainHint(request.domainHint);
         }
 
+        // Add sid or loginHint with preference for sid -> loginHint -> username of AccountInfo object
         if (request.sid) {
             parameterBuilder.addSid(request.sid);
+        } else if (request.loginHint) {
+            parameterBuilder.addLoginHint(request.loginHint);
+        } else if (request.account && request.account.username) {
```

Comment:
`Account` can also have `sid`? Used to in msal-core, not sure how relevant it is for browser!

Reply:
We should use `sid` if it is present in the id token.

Reply:
Then, we need to account for `sid` in `account`

Reply:
That's a good suggestion but will take it in a follow up PR

Reply:
+1 for a follow up PR